frb/analysis/kcwi.py

```diff
@@ -0,0 +1,362 @@
+        fill_value (array-like, optional): input to scipy.interpolate.interp1d
+            Check out this link for details
+            https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.interp1d.html
+        **readkw: Keyword arguments for reading the
+            input file using astropy Table.
+    Returns:
+        transfunc (function): interpolation
+    """
+    wave, transval = Table.read(transfile, **readkw)
+    trans = interp1d(wave, transval, kind=kind, fill_value=fill_value)
+    return trans
+
+def get_img(cubefile, wlow = None, whigh = None, trans_curve = None, save = None, overwrite = False):
+    """
+    Flatten cube along wavelength and produce a 2D
+    image.
+    Args:
+        cubefile (str): Path to the datacube
+        wlow, whigh (Quantity, optional): wavelength 
+            limits (with astropy units) to flatten between.
+            If nothing is given, the cube is checked for the
+            WAVGOOD keywords and flattened between
+            them. If they don't exist, it's flattened fully.
+        filter (function, optional): transmission
+            curve as a function of wavelength. Should be able
+            to take vector inputs and produce vector outputs.
+            We recommend passing a function produced by scipy
+            interpolation method. Wavelength is assumed to
+            be in angstroms.
+        save (str, optional): Path to file to be
+            saved to.
+        overwrite (bool, optional): Overwrite existing
+            file?
+    Returns:
+        img (Spectral Cube Projection): Flattened 2D image
+    """
+    # Read in datacube
+    cube = SpectralCube.read(cubefile)
+
+    # Create a truncated cube based on wlow and whigh
+    if not wlow:
+        try:
+            wlow = cube.header['WAVGOOD0']*cube.spectral_axis.unit
+        except KeyError:
+            wlow = cube.spectral_extrema[0]
+    
+    if not whigh:
+        try:
+            whigh = cube.header['WAVGOOD1']*cube.spectral_axis.unit
+        except KeyError:
+            whigh = cube.spectral_extrema[1]
+    
+    goodcube = cube.spectral_slab(wlow, whigh)
+
+    # Do you want to use a filter?
+    if trans_curve:
+        # Compute transmission curve for cube wavelengths
+        trans = trans_curve(goodcube.spectral_axis.value)
+
+        # Create a 3D array of trans stacked in the same
+        # shape as the cube spatial dimensions.
+        # TODO: make this more elegant.
+        tt = _spectral_tile(trans, cube)
+        goodcube = goodcube*tt
+    
+    # Make image
+    img = goodcube.sum(axis = 0, how = "ray")
+    if save:
+        img.write(save, overwrite = overwrite)
+    
+    return img
+
+def spec_from_mask(cube, mask_arr, varcube=None, kind="mean"):
+    """
+    Extract a spectrum from a cube
+    within a mask.
+    Args:
+        cube (Spectral Cube): A datacube object
+        mask_arr (numpy array): A 2D boolean array
```

Comment:
Is this mask intended to identify which pixels are extracted?
I think so.  Please specify.

Reply:
Yes. Changed now.